### PR TITLE
bigquery: read query results only if query finished

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

**Comment by pongad** [comment]
Tim, this addresses your concern from #2593.

I have one concern though: wouldn't it be easier to just make `Job.getQueryResults` wait for the job to complete? Even when we move to `tabledata.list`, we'll still need to call `getQueryResults` RPC to read the schema of the table.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java [hunk 1]
```diff
@@ -268,12 +267,14 @@ public QueryResult getQueryResults(QueryResultsOption... options) {
...
     }
-
-    // TODO(pongad): check job stats.state instead.
-    QueryResponse response = bigquery.getQueryResults(getJobId(), options);
-    if (!response.jobCompleted()) {
-      throw new IllegalStateException("the job hasn't completed yet");
+    if (getStatus().getState() != JobStatus.State.DONE) {
+      throw new BigQueryException(BigQueryException.UNKNOWN_CODE, "the job hasn't completed yet");
     }
+    if (getStatus().getError() != null) {
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: I assume `getError()` corresponds to [job.status.errorResult](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#status.errorResult)?

Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: Yes it is.

**Comment by pongad** [comment]
@tswast I thought about this a little more. Making `getQueryResults()` (the Java method, not RPC) wait for response can be a little confusing, since the `Job` won't be updated to reflect the fact that the query has already finished.

I've noted this down in #2591, but I'll submit this for now. It's in a branch so we can change our minds later.

**Pull request merged**